docs/design/fabric-v2+/fpc-registration.puml

```diff
@@ -12,261 +12,155 @@
   note left
     Org_1 takes "ownership" of this enclave
-    sign <CC_NAME, CC_VER, Org_1, ECC_S_PK, ECC_E_PK> with Org_1_S_SK
   end note
   return SIG_Org_1
-  |||
+  end group
 
   Enclave1 -> Enclave1 : generate quote
   note left
     quote is defined with reportbody =
-    Hash(<CC_NAME, CC_VER, Org_1, SIG_Org_1, ECC_S_PK, ECC_E_PK, TLCC_MR, CH_id>)
+    Hash(<CC_Id, CC_Version, MSPID, Enclave_VK, Enclave_EK, CC_EK, TLCC_MR, Channel_Id>)
   end note
 
-  Enclave1 -> Enclave1 : seal state
+  Enclave1 -> Enclave1 : populate protobuf
   note left
-    sealed state includes ...
-    CC_NAME, CC_VER,
-    Org_1, SIG_Org_1,
-    ECC_S_PK, ECC_S_SK,
-    ECC_E_PK, ECC_E_SK,
-    TLCC_MR, CH_id
+    attestation and (public) keys are inserted in
+    the Credentials protobuf
   end note
 
-  return ECC_S_PK, ECC_E_PK, sealed state, quote
-
-  return ECC_S_PK, ECC_E_PK, sealed state, quote, SIG_Org_1
-
-  |||
-
-  Manager1 -> storage1 ++ : store sealed state
-  note right
-    sealed state is stored under a
-    unique identifier consisting of
-    CC_NAME, CC_VER, CC_ID, ECC_S_PK, ECC_E_PK
-  end note
-  return
-
-  create entity IAS order 22
-  Manager1 -> IAS ++: send quote
-  note right
-    use IAS credentials and url provided by config
-  end note
-  note over Manager1 #FFaa00
-    **MST**: From an attestation generalization perspective i think there should be just a single
-    call ""GetAttestationProof"" or alike rather than a separate place where you get quote
-    and another where you get the registration report which requires some re-organization of calls.
-    A few observations on this:
-    - we do not strictly have to call the enclave via ECC. ECCManager could also directly call
-      enclave.so. this would make this creation and key-management potentially easier?
-    - the quote generation is not driven from within the enclave as currently shown above but from
-      the outside!
-    I think from a high-level there should be following three high-level calls either in
-    ECC or ECCMAnager depending on above:
-    - EnclaveCreate (key-gen & imprint of context, i.e., current create/binding flows
-    - EnclaveProof: gets org endorsement-sig, creates report, then quote
-      and, if epid, gets IAS report. 
-    - EnclavePersist: returns sealed state (important for security that this is called only after
-      EnclaveProof [which we might want to enforce using some state-machine?] )
-  end note
-  return ATTESTATION
-
-  |||
-
-  Manager1 -> Manager1 : mark ECC_ID as installed
-  note right
-    Manager1 maintains a list of
-    locally "installed/created" enclaves
+  group Persistent Credentials (to be discussed)
+  Enclave1 -> Enclave1 : seal state
+  note left
+    sealed state includes: ...
+    CC_Id, CC_Version,
+    MSPID, Channel_Id,
+    Enclave_SK, Enclave_DK, CC_DK,
+    TLCC_MR
   end note
-  return ATTESTATION, ECC_S_PK / error
-
-  deactivate Admin
-  |||
+  end group
+
+  return Credentials protobuf
```

Comment:
See above why i think below should be done peer-side and not client-side.  See also the related high-level calls i've outline in my comment around attestation in the old diagram ...